lib/numerizer.rb

```diff
@@ -108,50 +105,105 @@ class Numerizer
     ['ninetieth', '90']
   ]
 
-  def self.numerize(string)
+  ALL_ORDINALS = ORDINALS + SINGLE_ORDINAL_FRACTIONALS + DIRECT_ORDINAL_FRACTIONALS
+  ALL_FRACTIONS = FRACTIONS + (SINGLE_ORDINAL_FRACTIONALS + DIRECT_ORDINAL_FRACTIONALS).map {|tp| [tp[0] + '(s)?', tp[1]] }
+  ONLY_PLURAL_FRACTIONS = FRACTIONS + (SINGLE_ORDINAL_FRACTIONALS + DIRECT_ORDINAL_FRACTIONALS).map {|tp| [tp[0] + 's', tp[1]] }
+
+  ALL_ORDINALS_REGEX = ALL_ORDINALS.reduce { |a,b| a[0] + '|' + b[0] }
+
+
+  def self.numerize(string, ignore: [], bias: :none)
     string = string.dup
+    ignore = ignore.map(&:downcase).to_set
 
     # preprocess
     string.gsub!(/ +|([^\d])-([^\d])/, '\1 \2') # will mutilate hyphenated-words
     string.gsub!(/\ba$/, '') && string.rstrip! # doesn't make sense for an 'a' at the end to be a 1
 
     # easy/direct replacements
     (DIRECT_NUMS + SINGLE_NUMS).each do |dn|
-      string.gsub!(/(^|\W)#{dn[0]}(?=$|\W)/i, '\1<num>' + dn[1].to_s)
+      next if ignore.include? dn[0].downcase 
+      string.gsub!(/(^|\W)#{dn[0]}(?=$|\W)/i, '\1<num>' + dn[1].to_s) 
     end
 
     # ten, twenty, etc.
     TEN_PREFIXES.each do |tp|
+      next if ignore.include? tp[0].downcase 
       SINGLE_NUMS.each do |dn|
+        next if ignore.include? dn[0].downcase 
```

Comment:
I'm not a fan of these ignore checks all over the place, it seems easy to miss one if the code changes. I don't have a great suggestion though :/.

It also adds a lot of extra loops to run through which isn't super efficient.